src/environment/mod.rs

```diff
@@ -251,11 +271,19 @@ impl Environment {
         }
     }
 
-    pub fn has_public_initialiser(&mut self, type_id: &str) -> bool {
+    pub fn get_public_initialiser(&mut self, type_id: &str) -> Option<&mut SpecialDeclaration> {
         self.types
             .get_mut(type_id)
             .unwrap()
             .public_initializer
+            .as_mut()
+    }
+
+    pub fn has_public_initialiser(&self, type_id: &str) -> bool {
```

Comment:
Is this used anywhere where pattern matching on `.get_public_initialiser` would be preferable?